testprojects/src/python/native/BUILD

```diff
@@ -2,18 +2,30 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 python_library(
+    name="lib",
+    sources=["name.py", "setup.py"],
 )
 
 resources(name="impl", sources=["*.c"])
 
 python_distribution(
     name = "dist",
-    dependencies = [":impl", ":native"],
-    provides = setup_py(
+    dependencies = [":impl", ":lib"],
+    provides = python_artifact(
```

Comment:
Why?

Reply:
I think we should deprecate `setup_py` as the object name. It's a lot less intuitive than `python_artifact`.

Reply:
Better is https://github.com/pantsbuild/pants/issues/12410